src/middleware/error-interceptors/dataIntegrityErrorInterceptor.ts

```diff
@@ -0,0 +1,24 @@
+import { NextFunction, Request, Response } from "express";
+import { DataIntegrityError, DataIntegrityErrorType } from "../../lib/errors/dataIntegrityError";
+import { HttpStatusCode } from "axios";
+import { PrefixedUrls, STOP_TYPE } from "../../constants";
+import { logger } from "../../lib/logger";
+import { getUrlWithStopType } from "../../utils/url";
+
+export function dataIntegrityErrorInterceptor (err: Error | DataIntegrityError, req: Request, res: Response, next: NextFunction): void {
+    if (!(err instanceof DataIntegrityError)) {
+        return next(err);
+    }
+    logger.error(`${err.name} (${err.type}): ${err.stack}`);
+
+    switch (err.type) {
```

Comment:
Update switch statement to if/else instead - as SonarQube code smell reported for switch statement with < 3 cases